src/hermes/commands/deposit/invenio.py

```diff
@@ -17,648 +17,613 @@
 
-    _log = logging.getLogger("cli.deposit.invenio")
+        communities = self._get_community_identifiers()
+        self.ctx.update(self.invenio_context_path["communities"], communities)
 
-    publish_url_path = ContextPath.parse("deposit.invenio.links.publish")
-    publish_url = ctx[publish_url_path]
+        access_right, embargo_date, access_conditions = self._get_access_modalities(license)
+        self.ctx.update(self.invenio_context_path["access_right"], access_right)
+        self.ctx.update(self.invenio_context_path["embargo_date"], embargo_date)
+        self.ctx.update(self.invenio_context_path["access_conditions"], access_conditions)
 
-    response = requests.post(
-        publish_url,
-        headers={
-            "User-Agent": hermes_user_agent,
-            "Authorization": f"Bearer {click_ctx.params['auth_token']}"
-        }
-    )
+        self.invenio_ctx = self.ctx[self.invenio_context_path]
 
-    if not response.ok:
-        _log.debug(response.text)
-        raise RuntimeError(f"Could not publish deposit via {publish_url!r}")
+    def map_metadata(self) -> None:
+        """Map the harvested metadata onto the Invenio schema."""
 
-    record = response.json()
-    _log.info("Published record: %s", record["links"]["record_html"])
+        deposition_metadata = self._codemeta_to_invenio_deposition()
+        self.ctx.update(self.invenio_context_path["depositionMetadata"], deposition_metadata)
 
+        # Store a snapshot of the mapped data within the cache, useful for analysis, debugging, etc
+        with open(self.ctx.get_cache("deposit", self.platform_name, create=True), 'w') as invenio_json:
+            json.dump(deposition_metadata, invenio_json, indent='  ')
 
-def _resolve_latest_invenio_id(ctx: CodeMetaContext) -> t.Tuple[str, dict]:
-    """
-    Using the given configuration and metadata, figure out the latest record id.
+    def is_initial_publication(self) -> bool:
+        latest_record_id = self.invenio_ctx.get("latestRecord", {}).get("id")
+        return latest_record_id is None
 
-    If a record id is present as configuration ``deposit.invenio.record_id`` this one will be used to identify the
-    latest version of the record. Otherwise, if there is a doi present (either as configuration with key
-    ``deposit.invenio.doi``  or as a codemeta identifier, the DOI will be used to resolve the base record id.
+    def create_initial_version(self) -> None:
+        """Create an initial version of a publication."""
 
-    Anyway, the record id will always be used to resolve the latest version.
+        if not self.click_ctx.params['initial']:
+            raise RuntimeError("Please use `--initial` to make an initial deposition.")
 
-    If any of the resolution steps fail or produce an unexpected result, a ValueError will be thrown.
+        response = self.client.new_deposit()
+        if not response.ok:
+            raise RuntimeError(f"Could not create initial deposit {response.url!r}")
```

Comment:
**chore:** The answer returned by the server is very important here and should be either logged or be part of the exception message.